llvm/lib/CodeGen/MachinePassManager.cpp

```diff
@@ -71,7 +73,9 @@ bool MachineFunctionAnalysisManagerModuleProxy::Result::invalidate(
 
 PreservedAnalyses
 ModuleToMachineFunctionPassAdaptor::run(Module &M, ModuleAnalysisManager &AM) {
-  auto &MMI = AM.getResult<MachineModuleAnalysis>(M).getMMI();
+  // Ensure we have a MachineModuleInfo
+  AM.getResult<MachineModuleAnalysis>(M).getMMI();
```

Comment:
is this temporary (if so add a TODO)? is it too much to remove `MachineModuleAnalysis` from the various APIs in this patch as well?